frontend/src/pages/MetricApplication/index.tsx

```diff
@@ -1,76 +1,89 @@
-import MetricsApplicationContainer from 'container/MetricsApplication';
-import useResourceAttribute from 'hooks/useResourceAttribute';
-import { convertRawQueriesToTraceSelectedTags } from 'hooks/useResourceAttribute/utils';
-import { useEffect, useMemo } from 'react';
-import { connect, useSelector } from 'react-redux';
-import { useParams } from 'react-router-dom';
-import { bindActionCreators } from 'redux';
-import { ThunkDispatch } from 'redux-thunk';
-import {
-	GetInitialData,
-	GetInitialDataProps,
-} from 'store/actions/metrics/getInitialData';
-import { AppState } from 'store/reducers';
-import AppActions from 'types/actions';
-import { GlobalReducer } from 'types/reducer/globalTime';
-import MetricReducer from 'types/reducer/metrics';
-import { Tags } from 'types/reducer/trace';
+import RouteTab from 'components/RouteTab';
+import ROUTES from 'constants/routes';
+import ResourceAttributesFilter from 'container/ResourceAttributesFilter';
+import history from 'lib/history';
+import { useMemo } from 'react';
+import { generatePath, useLocation, useParams } from 'react-router-dom';
 
-function MetricsApplication({ getInitialData }: MetricsProps): JSX.Element {
-	const { minTime, maxTime } = useSelector<AppState, GlobalReducer>(
-		(state) => state.globalTime,
-	);
-	const { error, errorMessage, metricsApplicationLoading } = useSelector<
-		AppState,
-		MetricReducer
-	>((state) => state.metrics);
+import DBCall from '../../container/MetricsApplication/Tabs/DBCall';
+import External from '../../container/MetricsApplication/Tabs/External';
+import Overview from '../../container/MetricsApplication/Tabs/Overview';
 
-	const { servicename } = useParams<ServiceProps>();
-	const { queries } = useResourceAttribute();
+function OverViewTab(): JSX.Element {
+	return <Overview />;
+}
 
-	const selectedTags = useMemo(
-		() => (convertRawQueriesToTraceSelectedTags(queries) as Tags[]) || [],
-		[queries],
-	);
+function DbCallTab(): JSX.Element {
+	return <DBCall />;
+}
 
-	useEffect(() => {
-		if (servicename !== undefined) {
-			getInitialData({
-				serviceName: servicename,
-				maxTime,
-				minTime,
-				selectedTags,
-			});
-		}
-	}, [servicename, getInitialData, maxTime, minTime, selectedTags]);
+function ExternalTab(): JSX.Element {
+	return <External />;
+}
 
-	if (metricsApplicationLoading) {
-		return <Spinner tip="Loading..." />;
-	}
+function MetricsApplication(): JSX.Element {
+	const { search } = useLocation();
+	const { servicename } = useParams<{ servicename: string }>();
 
-	if (error) {
-		return <Typography>{errorMessage}</Typography>;
-	}
+	const searchParams = new URLSearchParams(search);
+	const tab = searchParams.get('tab');
 
-	return <MetricsApplicationContainer />;
-}
+	const overMetrics = 'Overview Metrics';
+	const dbCallMetrics = 'Database Calls';
+	const externalMetrics = 'External Calls';
 
-interface DispatchProps {
-	getInitialData: (props: GetInitialDataProps) => void;
-}
+	const getActiveKey = (): string => {
```

Comment:
const getActiveKey = (): string => {
    const metricsMap = {
        [dbCallMetrics]: dbCallMetrics,
        [externalMetrics]: externalMetrics,
    };

    return metricsMap[tab] || overMetrics;
};

Reply:
But the `tab` can be null, then it will be like 

```ts
const getActiveKey = (): string => {
	const metricsMap = {
		[dbCallMetrics]: dbCallMetrics,
		[externalMetrics]: externalMetrics
	}
	if (tab) {
		return metricsMap[tab] || overMetrics
	} else {
		return overMetrics
	}
}
```

Will this be better, performance vise both approaches have similar performance in terms of execution time.

Reply:
default case is already handled by `||`

Reply:
Okay